tests/utils/test_rest_utils.py

```diff
@@ -820,3 +819,107 @@ def mock_failing_call():
     actual_sleep_times = [call.args[0] for call in mock_sleep.call_args_list]
     assert actual_sleep_times == expected_sleep_times
     assert call_count == 4  # Initial + 3 retries
+
+
+@mock.patch("databricks.sdk.WorkspaceClient")
+@mock.patch("databricks.sdk.config.Config")
+def test_timeout_parameter_propagation_with_timeout(mock_config, mock_workspace_client):
+    """Test timeout parameter propagation from http_request to get_workspace_client with timeout."""
+    from mlflow.utils.rest_utils import get_workspace_client
+
+    # Test http_request with timeout via get_workspace_client directly
+    mock_workspace_client_instance = mock.MagicMock()
+    mock_workspace_client_instance.api_client.do.return_value = {"contents": mock.MagicMock()}
+    mock_workspace_client.return_value = mock_workspace_client_instance
+
+    get_workspace_client(
+        use_secret_scope_token=False,
+        host="http://my-host",
+        token=None,
+        databricks_auth_profile="my-profile",
+        retry_timeout_seconds=None,
+        timeout=180,
+    )
+
+    # Verify Config was called with profile and http_timeout_seconds
+    expected_kwargs = {
+        "profile": "my-profile",
+        "http_timeout_seconds": 180,
+    }
+    mock_config.assert_called_once_with(
+        **expected_kwargs,
+        retry_timeout_seconds=mock.ANY,
+    )
+
+
+@mock.patch("databricks.sdk.WorkspaceClient")
+@mock.patch("databricks.sdk.config.Config")
+def test_timeout_parameter_propagation_without_timeout(mock_config, mock_workspace_client):
+    """Test timeout parameter propagation from http_request to get_workspace_client without timeout."""
+    from mlflow.utils.rest_utils import get_workspace_client
+
+    # Test http_request without timeout via get_workspace_client directly
+    mock_workspace_client_instance = mock.MagicMock()
+    mock_workspace_client_instance.api_client.do.return_value = {"contents": mock.MagicMock()}
+    mock_workspace_client.return_value = mock_workspace_client_instance
+
+    get_workspace_client(
+        use_secret_scope_token=False,
+        host="http://my-host",
+        token=None,
+        databricks_auth_profile="my-profile",
+        retry_timeout_seconds=None,
+        timeout=None,
+    )
+
+    # Verify Config was called with profile and no http_timeout_seconds
+    expected_kwargs = {
+        "profile": "my-profile",
+    }
+    mock_config.assert_called_once_with(
+        **expected_kwargs,
+        retry_timeout_seconds=mock.ANY,
+    )
+
+
+@mock.patch("mlflow.utils.rest_utils.get_workspace_client")
+@mock.patch("mlflow.utils.databricks_utils.get_databricks_host_creds")
```

Comment:
can we use a context manager? decorators makes it difficult to guess what needs `mock.patch`.